src/api.js

```diff
@@ -1,150 +1,145 @@
-import { BASE_URL } from "./constants";
-import { clearContent, notification } from "./util.dom";
-import KEY from "../key";
+import { BASE_URL } from './constants';
+import { clearContent, notification } from './util.dom';
+import KEY from '../key';
 
 const apiModule = () => {
+  const handleError = (err) => {
+    console.warn('Error', err);
+    document.querySelector('body').appendChild(notification(`Error: ${err.statusText}`));
+    setTimeout(() => {
+      clearContent('[id="notify"', true);
+    }, 2300);
+  };
 
-    const handleError = (err) => {
-        console.warn('Error', err);
-        document.querySelector('body').appendChild(notification('Network Error'));
-        setTimeout(() => {
-            clearContent('[id="notify"', true);
-        }, 2300);
-    }
-
-    const post = async (uri = '', data = {}) => {
-        const response = await fetch(`${BASE_URL + uri}&appid=${KEY}`, {
-            method: 'POST',
-            mode: 'cors',
-            cache: 'no-cache',
-            credentials: 'same-origin',
-            headers: {
-                'Content-Type': 'application/json'
-            },
-            redirect: 'follow',
-            referrerPolicy: 'no-referrer',
-            body: JSON.stringify(data)
-        }).then(function (response) {
-            if (response.ok) {
-                return response.json();
-            } else {
-                return Promise.reject(response);
-            }
-        }).catch(handleError);
-
-        if (response) {
-            return response;
-        }
-        return false;
-    };
+  const post = async (uri = '', data = {}) => {
+    const response = await fetch(`${BASE_URL + uri}&appid=${KEY}`, {
+      method: 'POST',
+      mode: 'cors',
+      cache: 'no-cache',
+      credentials: 'same-origin',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      redirect: 'follow',
+      referrerPolicy: 'no-referrer',
+      body: JSON.stringify(data),
+    }).then((response) => {
+      if (response.ok) {
+        return response.json();
+      }
+      return Promise.reject(response);
+    }).catch(handleError);
```

Comment:
- [ ] [REQUIRED] Here you are mixing concepts, if you are already using await, you don't need the `.then` at the end. Also if you use `async/await` you need to wrap your login inside of a `try/catch` statement to handle errors
![image](https://user-images.githubusercontent.com/73254040/101089704-27358300-357b-11eb-8b15-1336f8d7680e.png)
[Source](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/)

Reply:
This comment is not addressed. Kindly address it